core/src/main/scala/org/http4s/blaze/pipeline/stages/SSLStage.scala

```diff
@@ -135,11 +135,15 @@ final class SSLStage(engine: SSLEngine, maxWrite: Int = 1024 * 1024)
   // cleans up any pending requests
   private[this] def handshakeFailure(t: Throwable): Unit = {
     val start = System.nanoTime
-    val results = handshakeQueue.result(); handshakeQueue.clear();
-    results.foreach {
-      case DelayedRead(_, p) => p.failure(t)
-      case DelayedWrite(_, p) => p.failure(t)
-    }
+    val results = handshakeQueue.result()
+    handshakeQueue.clear()
+    if (results.isEmpty)
+      logger.error(t)(s"Handshake failure with no pending results")
```

Comment:
I'm not sure that this should ever happen, but we shouldn't be silent about it if it does.